lib/magic/cards/containment_priest.rb

```diff
@@ -0,0 +1,21 @@
+module Magic
+  module Cards
+    ContainmentPriest = Creature("Containment Priest") do
+      cost generic: 1, white: 1
+      creature_type("Human Cleric")
+      keywords :flash
+      power 2
+      toughness 2
+
+      def event_handlers
+        {
+          Events::EnteredTheBattlefield => -> (receiver, event) do
+            return if receiver == event.permanent
+            return if game.current_turn.events[-3].is_a? Magic::Events::SpellCast
```

Comment:
this is a slightly brittle implementation here and I'm including a magic number, but I wasn't sure how else I might best access that same information. Would be happy for an additional suggestion!

Reply:
I think the entered the battlefield event should include information about whether that was triggered by a spell cast or not. Then use that info here.

Reply:
Thanks for the fast review! So, I looked into `Events::EnteredTheBattlefield`, as well as `ETB`/`TriggeredAbility::EnterTheBattlefield` (though the latter seemed, upon further reflection, to be focused on when the Creature enters the Battlefield, rather than on responding to ETB events in general). 

Unfortunately, when I looked through the methods of `Events::EnteredTheBattlefield`, I couldn't find any methods that would identify the provenance of how the ETB was triggered. However, in an attempt to try to be a bit more robust in my approach, I refactored to rely on a different assumption: namely, that actions are a queue. 

If this holds, then the latest changes I pushed would rely on whether or not the latest action (if it exists, hence safe navigator operators) was, indeed, a `Cast` action. Hopefully this better satisfies code quality requirements as well as safety assumptions about how turns operate?

Reply:
Yes this'll do for now :) Let's not let perfect be the enemy of good.

I've added one small pending spec for handling a situation where a card can go from graveyard to battlefield, but since #129 has not been completed, I cannot finish writing that test. I think Containment Priest might handle that already.

Another situation to think of is where a stack reading top-to-bottom is:

1. Shock
2. Rise Again

Shock (#169) would resolve first, dealing two damage. At this point, I think Shock would've been the last action taken in the game. Then Rise Again works, and since that's the 2nd to last action, Containment Priest will not exile the card that is transitioning from Graveyard to Battlefield. I hope that has explained my thinking about that pending spec there.